### PR TITLE
Add support for Azure Disk storageClass

### PR DESCRIPTION
**Pull request by oprinmarius**
**What type of PR is this?**

/kind feature

**What this PR does / why we need it**:
* Adds support for Azure-Disk storageClass

**Which issue(s) this PR fixes**:

part of https://github.com/kubernetes/perf-tests/issues/2035

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: clusterloader2/pkg/prometheus/manifests/0ssd-storage-class.yaml [hunk 1]
```diff
@@ -7,7 +7,18 @@ metadata:
   name: ssd
 provisioner: {{$PROMETHEUS_STORAGE_CLASS_PROVISIONER}}
 parameters:
+  {{if eq $PROMETHEUS_STORAGE_CLASS_PROVISIONER "kubernetes.io/gce-pd"}}
```
Comment (mborsz) [inline clusterloader2/pkg/prometheus/manifests/0ssd-storage-class.yaml:10]: This stops setting type for PROMETHEUS_STORAGE_CLASS_PROVISIONER != kubernetes.io/gce-pd which is a breaking changes and will break some gke tests.

Can you remove this if and set PROMETHEUS_STORAGE_CLASS_VOLUME_TYPE to empty value when needed?
IIUC empty value shouldn't get overridden by DefaultParam above: https://github.com/kubernetes/perf-tests/blob/d8d8c4b6016376f3871432dd221e44649bb73837/clusterloader2/pkg/config/template_functions.go#L202-L202.

Comment (oprinmarius) [reply clusterloader2/pkg/prometheus/manifests/0ssd-storage-class.yaml:10]: `type` is not a valid parameter for azure-disk storageClass and must be replaced with other parameters. I will update this

**Comment by marosset** [comment]
Can the azuredisk drivers be deployed with either https://github.com/kubernetes-sigs/azuredisk-csi-driver/tree/master/charts/v1.15.0 or https://github.com/kubernetes-sigs/azuredisk-csi-driver/blob/master/docs/install-csi-driver-v1.15.0.md ?

**Pull request merged**